modules/sdk-core/src/bitgo/lightning/lightning.ts

```diff
@@ -114,4 +119,33 @@ export class Lightning implements ILightning {
       throw new Error(`error(s) parsing response body: ${errors}`);
     });
   }
+
+  public async decodeLnurlPay(lnurl: string): Promise<LnurlPayResponse> {
+    const { prefix, words } = bech32.decode(lnurl, 2000);
+
+    if (prefix !== 'lnurl') {
+      throw new Error('invalid lnurl');
+    }
+
+    const url = Buffer.from(bech32.fromWords(words)).toString();
+    const body = await this.bitgo.get(url).result();
+    const decodedRes = decodeOrElse(LnurlPayResponse.name, LnurlPayResponse, body, (errors) => {
+      throw new Error(`error(s) parsing lnurl response: ${errors}`);
+    });
+    const { availableBalance } = await this.getBalance();
+    const domain = new URL(url).hostname.replace('www.', '');
```

Comment:
why?

Reply:
From https://github.com/fiatjaf/lnurl-rfc/blob/luds/06.md#pay-to-static-qrnfclink

> Additionally, a payment dialog must include:
>
> - Domain name extracted from LNURL query string.

This would be good to reference in a comment though.

Reply:
lnurl protocol specifies that we should provide domain name extracted from lnurl query string